cluster-autoscaler/processors/nodegroupset/compare_nodegroups.go

```diff
@@ -110,14 +113,29 @@ func IsCloudProviderNodeInfoSimilar(n1, n2 *schedulernodeinfo.NodeInfo, ignoredL
 			free[res] = append(free[res], freeRes)
 		}
 	}
-	// For capacity we require exact match.
-	// If this is ever changed, enforcing MaxCoresTotal and MaxMemoryTotal limits
-	// as it is now may no longer work.
-	for _, qtyList := range capacity {
-		if len(qtyList) != 2 || qtyList[0].Cmp(qtyList[1]) != 0 {
+
+	for kind, qtyList := range capacity {
+		if len(qtyList) != 2 {
 			return false
 		}
+		switch kind {
+		case apiv1.ResourceMemory:
+			// For memory capacity we allow a small tolerance
+			larger := math.Max(float64(qtyList[0].Value()), float64(qtyList[1].Value()))
+			smaller := math.Min(float64(qtyList[0].Value()), float64(qtyList[1].Value()))
+			if larger-smaller > MaxMemoryDifferenceInKiloBytes {
```

Comment:
```
memoryDifference := Math.Abs(qtyList[0].Value()) - float64(qtyList[1].Value())
if memoryDifference >
```

Reply:
Done in https://github.com/kubernetes/autoscaler/compare/56d99b72b3384b20b33437571e605eb644cf5172..e8b3c2a111eb9b3b4fe15b4d4081175267ff6d76